internal/dispatch/cluster/cluster.go

```diff
@@ -0,0 +1,54 @@
+package cluster
+
+import (
+	"github.com/dgraph-io/ristretto"
+
+	"github.com/authzed/spicedb/internal/dispatch"
+	"github.com/authzed/spicedb/internal/dispatch/caching"
+	"github.com/authzed/spicedb/internal/dispatch/graph"
+	"github.com/authzed/spicedb/internal/namespace"
+)
+
+// Option is a function-style option for configuring a combined Dispatcher.
+type Option func(*optionState)
+
+type optionState struct {
+	prometheusSubsystem string
+	cacheConfig         *ristretto.Config
+}
+
+// PrometheusSubsystem sets the subsystem name for the prometheus metrics
+func PrometheusSubsystem(name string) Option {
+	return func(state *optionState) {
+		state.prometheusSubsystem = name
+	}
+}
+
+// CacheConfig sets the configuration for the local dispatcher's cache.
+func CacheConfig(config *ristretto.Config) Option {
+	return func(state *optionState) {
+		state.cacheConfig = config
+	}
+}
+
+// NewClusterDispatcher takes a caching redispatcher (such as one created by
+// combined.NewDispatcher) and returns a cluster dispatcher suitable for use as
+// the dispatcher for the dispatch grpc server.
+func NewClusterDispatcher(cachingRedispatch dispatch.Dispatcher, nsm namespace.Manager, options ...Option) (dispatch.Dispatcher, error) {
```

Comment:
Should the internal dispatcher be called `cachingDispatcher`? It doesn't *have* to be cached, right?